UI/CUIControls.h

```diff
@@ -594,4 +595,48 @@ class RotatingGraphic : public GG::StaticGraphic {
     float   m_phase_offset;
 };
 
+class CUIToggleButton;
+
+/** \brief A FreeOrion styled ToggleButton. */
+class CUIToggleButtonRepresenter : public GG::StateButtonRepresenter {
+public:
+    virtual void   Render(const CUIToggleButton& button) const;
+
+    virtual void   OnChecked(bool checked) const;
+
+    virtual GG::Pt MinUsableSize(const CUIToggleButton& button) const;
+};
+
+/** \brief Toggled StateButton
+ * 
+ * This StateButton renders a SubTexture depending on the checked state.
+ * With the default single texture it renders the same icon recolored towards grey.
+ * When a second SebTexture is assigned, it will render the appropriate SubTexture.
+ */
+class CUIToggleButton : public GG::StateButton {
```

Comment:
[GG/GG/Button.h:133](https://github.com/freeorion/freeorion/blob/master/GG/GG/Button.h#L133)
>    The StateButtonRepresenter is a stub interface to implement a visual
    representation of a StateButton instance.  A representer can be shared
    amongst multiple instances of a StateButton, so no subclass should
    store instance specific attributes of a state button.

I realize this is your code, but this is where I came to the conclusion not to include the references within CUIToggleButtonRepresenter.